security/pfSense-pkg-snort/files/usr/local/www/snort/snort_ip_reputation.php

```diff
@@ -35,26 +35,16 @@
 	exit;
 }
 
-$a_nat = &$config['installedpackages']['snortglobal']['rule'];
+$a_nat = config_get_path("installedpackages/snortglobal/rule/{$id}", []);
 
-$pconfig = $a_nat[$id];
-$iprep_path = SNORT_IPREP_PATH;
-$if_real = get_real_interface($a_nat[$id]['interface']);
-$snort_uuid = $config['installedpackages']['snortglobal']['rule'][$id]['uuid'];
+// Init 'blist_files' and 'wlist_files' arrays for the interface
+array_init_path($a_nat, 'blist_files/item');
+array_init_path($a_nat, 'wlist_files/item');
 
-// Init 'blist_files' and 'wlist_files' arrays if necessary
-if (!is_array($a_nat[$id]['blist_files'])) {
-	$a_nat[$id]['blist_files'] = array();
-}
-if (!is_array($a_nat[$id]['blist_files']['item'])) {
-	$a_nat[$id]['blist_files']['item'] = array();
-}
-if (!is_array($a_nat[$id]['wlist_files'])) {
-	$a_nat[$id]['wlist_files'] = array();
-}
-if (!is_array($a_nat[$id]['wlist_files']['item'])) {
-	$a_nat[$id]['wlist_files']['item'] = array();
-}
+$pconfig = $a_nat;
```

Comment:
I don't quite understand the necessity of keeping this copy of $a_nat around, when at each change of $a_nat it looks as though $pconfig is assigned the same changes. It seems to me that $pconfig could be removed entirely in favor of $a_nat.

Reply:
The idea is to preserve certain changes on the page between post backs depending on what exactly the user clicked. There are two distinct types of things going on with that page. One is saving basic settings, and the other is adding or removing blacklist/whitelist files (text files of IP addresses) and assigning the files to either whitelists or blacklists. Looking back now, I don't recall of my reasoning at the time I created that page.